### DISCUSSION TITLE
[02_pytorch_classification.ipynb] | RuntimeError: mat1 and mat2 shapes cannot be multiplied (200x5 and 2x5)

### DISCUSSION THREAD

**Discussion by prince-kha** [mrdbourke/pytorch-deep-learning#232] [Q&A] [answered]
I was practicing the section, [02_pytorch_classification.ipynb](https://github.com/mrdbourke/pytorch-deep-learning/blob/main/docs/02_pytorch_classification.ipynb) and faced this issue. I used the same code from the google colab but the problem persists.

```python
# Make predictions with the model
untrained_preds = model_0(X_test.to(device))
print(f"Length of predictions: {len(untrained_preds)}, Shape: {untrained_preds.shape}")
print(f"Length of test samples: {len(Y_test)}, Shape: {Y_test.shape}")
print(f"\nFirst 10 predictions:\n{untrained_preds[:10]}")
print(f"\nFirst 10 test labels:\n{Y_test[:10]}")
```

------------------------------------------------------------------------------------------------------------

Error:
```
RuntimeError                              Traceback (most recent call last)
[<ipython-input-72-54cb01733233>](https://localhost:8080/#) in <module>
      1 # Make predictions with the model
----> 2 untrained_preds = model_0(X_test.to(device))
      3 print(f"Length of predictions: {len(untrained_preds)}, Shape: {untrained_preds.shape}")
      4 print(f"Length of test samples: {len(Y_test)}, Shape: {Y_test.shape}")
      5 print(f"\nFirst 10 predictions:\n{untrained_preds[:10]}")

3 frames
[/usr/local/lib/python3.8/dist-packages/torch/nn/modules/linear.py](https://localhost:8080/#) in forward(self, input)
    112 
    113     def forward(self, input: Tensor) -> Tensor:
--> 114         return F.linear(input, self.weight, self.bias)
    115 
    116     def extra_repr(self) -> str:

RuntimeError: mat1 and mat2 shapes cannot be multiplied (200x5 and 2x5)
```

**Comment by AlienSarlak** [comment]
As you can see the error is :

`RuntimeError: mat1 and mat2 shapes cannot be multiplied (200x5 and 2x5)`

meaning that the shape of your matrices are not correct. You cannot multiply two matrices that the number of **columns** in the **first** matrix is **not** equal to the number of **rows** in the **second** matrix

**Comment by mrdbourke** [comment, marked as answer]
Hi @prince-kha, 

@AlienSarlak is correct, the tensors you're trying to put into the network are incorrect shapes.

I'd check the shapes of all of your inputs and make sure they line up with the matrix multiplication rule: https://www.learnpytorch.io/00_pytorch_fundamentals/#matrix-multiplication-is-all-you-need
